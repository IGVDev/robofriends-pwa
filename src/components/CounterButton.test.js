import React from "react";
import CounterButton from "./CounterButton";
import { shallow, mount, render } from "enzyme";
import "../setupTests";

it("expect to render Card component", () => {
  const mockColor = "red";
  expect(
    shallow(<CounterButton color={mockColor} />).debug()
  ).toMatchSnapshot();
});

it("correctly increments the counter", () => {
  const mockColor = "red";
  const wrapper = shallow(<CounterButton color={mockColor} />);

  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 1 });
  wrapper.find('[id="counter"]').simulate("click");
  wrapper.find('[id="counter"]').simulate("click");
  expect(wrapper.state()).toEqual({ count: 3 });
  wrapper.find('[id="counter"]').simulate("keypress");
  expect(wrapper.state()).toEqual({ count: 3 });
  expect(wrapper.props().color).toEqual("red");
});
