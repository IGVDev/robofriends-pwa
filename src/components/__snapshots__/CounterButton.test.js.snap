// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
"<button id=\\"counter\\" color=\\"red\\" onClick={[Function (anonymous)]}>
  Count: 
  0
</button>"
`;
